src/main/java/com/opentable/metrics/JettyServerMetricsConfiguration.java

```diff
@@ -67,7 +83,17 @@
         };
     }
 
+    /**
+     * Instrumented Queued Thread Pool that removes request ID from the MDC after the job is run
+     */
     static class OTQueuedThreadPool extends InstrumentedQueuedThreadPool {
+
+        /**
+         * Create a queued thread pool. Handles request Id in MDC.
+         *
+         * @param metricRegistry the metric registry used to record metrics on pool utilization
+         * @param qSize the size of the request queue (used if there are no available threads) - THIS IS NOT USED, the queue is unbounded to avoid memory leaks
```

Comment:
"unbounded to avoid memory leaks" seems an oxymoron.  What am I missing?

Reply:
If it's bounded and we fill up the queue then we'll throw RejectedExecutionException's. Jetty leaks resources when that happens. We prefer to run out of memory rather than have the leak. (It's explained further in comments below.)